src/VisualStudio/Core/Def/Implementation/FindUsages/VisualStudioFindSymbolMonikerUsagesService.cs

```diff
@@ -106,9 +108,11 @@ private ExternalReferenceItem ConvertResult(DefinitionItem definition, JObject o
             var displayPath = obj.Value<string>("displayPath");
             var span = ConvertLinePositionSpan(obj.Value<JObject>("range"));
             var text = obj.Value<string>("text");
+            var repository = obj.Value<string>("repository");
+            var scope = (ExternalScope)obj.Value<int>("scope");
```

Comment:
can we just define a type with attributes for this rather than manually converting each field?  Then CodeIndexExternalReferenceItem could just be that type rather than passing in all the fields.

Reply:
1. i actually prefer manual deserialization because it makes it clear to me what's going on.
2. it's a very small amount of code.
3. all of this goes away post build because LSP will just contact rich-nav directly, without going through roslyn.  so i'm not really keen to invest further.